src/schedule/zephyr_ll.c

```diff
@@ -198,44 +199,51 @@ static void zephyr_ll_run(void *data)
 			continue;
 		}
 
-		pdata->run = true;
-		task->state = SOF_TASK_STATE_RUNNING;
+		if (domain_is_pending(domain, task, NULL)) {
+			pdata->run = true;
+			task->state = SOF_TASK_STATE_RUNNING;
+		}
 
 		/* Move the task to a temporary list */
 		list_item_del(list);
 		list_item_append(list, &task_head);
 
-		zephyr_ll_unlock(sch, &flags);
-
-		/*
-		 * task's .run() should only return either
-		 * SOF_TASK_STATE_COMPLETED or SOF_TASK_STATE_RESCHEDULE
-		 */
-		state = do_task_run(task);
-		if (state != SOF_TASK_STATE_COMPLETED &&
-		    state != SOF_TASK_STATE_RESCHEDULE) {
-			tr_err(&ll_tr,
-			       "zephyr_ll_run: invalid return state %u",
-			       state);
-			state = SOF_TASK_STATE_RESCHEDULE;
-		}
-
-		zephyr_ll_lock(sch, &flags);
+		if (task->state == SOF_TASK_STATE_RUNNING) {
```

Comment:
The diff is a bit hard to read but I can't spot any changes in the function. So could this be nicer to read with:

```
if (task->state != SOF_TASK_STATE_RUNNING)
     continue;
```

Reply:
> The diff is a bit hard to read but I can't spot any changes in the function. So could this be nicer to read with:
> 
> ```
**> if (task->state != SOF_TASK_STATE_RUNNING)
>      continue;**
This should better to read.
> ```

Hi @kv2019i , thanks for your review the patches.